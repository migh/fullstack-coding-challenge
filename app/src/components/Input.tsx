import * as React from 'react';

export interface InputProps {
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  onEnter: (event: React.FormEvent<HTMLInputElement>) => void;
  placeholder: string;
  value: string;
}

export const Input = (props: InputProps) => (
    <input
      className="form-control"
      type="text"
      placeholder={props.placeholder}
      value={props.value}
      onChange={props.onChange}
      onKeyPress={props.onEnter}
    />
);

