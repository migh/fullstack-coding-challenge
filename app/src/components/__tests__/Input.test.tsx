import * as React from 'react';
import { shallow } from 'enzyme';

import { Input } from '../Input';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should render.', () => {
      const props = {
        placeholder: 'Some text.',
        value: '',
        onChange: () => {},
        onEnter: () => {}
      };
      expect(shallow(<Input {...props} />)).toMatchSnapshot();
    });
  });
});
