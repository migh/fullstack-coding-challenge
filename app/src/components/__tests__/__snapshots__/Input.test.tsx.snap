// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should render. 1`] = `ShallowWrapper {}`;
